File: csharpcore/GildedRose/ItemTypes/LegendaryItemType.cs
namespace GildedRoseKata.ItemTypes;

public class LegendaryItemType : ItemType
{
  public LegendaryItemType()
  {
    Type = "Legendary";
    QualityModifier = 0;
  }

  // Legendary Items do not lose or gain Quality and have no sellIn timeframe. Values remain the same.
  public override void UpdateQuality(ref int quality, ref int sellIn) {}
}

File: ruby/ItemTypes/legendary_item_type.rb
require File.join(File.dirname(__FILE__), 'item_type')

class LegendaryItemType < ItemType
  def initialize
    @quality_modifier = 0
    @type = "Legendary"
    super
  end

  def update_quality(quality, sell_in)
    [quality,sell_in]
  end
end
